src/core/arrayAsSequence.js

```diff
@@ -0,0 +1,131 @@
+import {Sequence} from "./sequence";
+import {isArray, isInteger} from "../core/types";
+import {wrap} from "../core/wrap";
+
+export const ArraySequence = Sequence.extend({
+    summary: "Enumerate the contents of an array.",
+    supportsWith: [],
+    supportsAlways: [
+        "length", "left", "back", "index", "slice", "has", "get", "copy", "reset"
+    ],
+    overrides: [
+        "array", "newArray"
+    ],
+    docs: process.env.NODE_ENV !== "development" ? undefined : {
```

Comment:
Once we get this through, I'll continue investing a way to make `env` available as a type.